src/mscorlib/src/System/Span.cs

```diff
@@ -416,5 +436,67 @@ internal static unsafe void CopyTo<T>(ref T destination, ref T source, int eleme
                 }
             }
         }
+
+        internal static unsafe void Fill<T>(ref T destination, T value, int elementsCount)
+        {
+            if (!JitHelpers.ContainsReferences<T>())
+            {
+                int size = Unsafe.SizeOf<T>();
+
+                int copiedCount; // Fill up to copiedCount in first loop
+                if (size <= sizeof(int))
+                    copiedCount = elementsCount <= copyInLoopBufferSize / 4 * 3 ? elementsCount : copyInLoopBufferSize / 4 * 2;
+                else
+                    copiedCount = elementsCount <= copyInLoopBufferSize * 3 / size ? elementsCount : Math.Min(1, copyInLoopBufferSize * 2 / size);
+
+                for (int i = copiedCount - 1; i >= 0; i--)
+                    Unsafe.Add(ref destination, i) = value;
+
+                elementsCount -= copiedCount;
+                if (elementsCount <= 0)
+                    return;
+
+                var cashCount = Math.Min(1, cashBufferSize / size);
+
+                fixed (byte* pSource = &Unsafe.As<T, byte>(ref destination))
+                {
+                    while (elementsCount > 0)
+                    {
+                        int copyLen = Math.Min(Math.Min(copiedCount, elementsCount), cashCount);
+                        fixed (byte* pDestination = &Unsafe.As<T, byte>(ref Unsafe.Add(ref destination, copiedCount)))
+                        {
+#if BIT64
+                            Buffer.Memmove(pDestination, pSource, (ulong)copyLen * (ulong)size);
+#else
+                            Buffer.Memmove(pDestination, pSource, (uint)copyLen * (uint)size);
+#endif
+                        }
+                        copiedCount += copyLen;
+                        elementsCount -= copyLen;
+                    }
+                }
+            }
+            else
+            {
+                for (int i = elementsCount - 1; i >= 0; i--)
+                    Unsafe.Add(ref destination, i) = value;
+            }
+        }
+
+        internal static unsafe void Clear<T>(ref T destination, int elementsCount)
+        {
+            if (!JitHelpers.ContainsReferences<T>())
+            {
+                fixed (byte* pDestination = &Unsafe.As<T, byte>(ref destination))
+                {
+                    Buffer.ZeroMemory(pDestination, (long)elementsCount * Unsafe.SizeOf<T>());
```

Comment:
`Buffer.ZeroMemory` is pretty slow naïve implementation. It would be better to do this e.g. using `Unsafe.InitBlockUnaligned`. Note that Unsafe type in corelib is internal implementation detail so you do not need to block on the API approval to add it.